code-challenges.test.js

```diff
@@ -14,32 +18,74 @@
 
-const colors1 = ["purple", "blue", "green", "yellow", "pink"]
-// Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
-const colors2 = [
-  "chartreuse",
-  "indigo",
-  "periwinkle",
-  "ochre",
-  "aquamarine",
-  "saffron"
-]
-// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+//**UPDATED PSEUDOCODE: Welcome back, after confiding in my cowrokers/peers they recommended an easier method called .shift() which removed the first element in the array and explained the expect.arrayContaining(array) belongs into my test and not function as well to create NOT to equal expect.  Thank you for reading, see you on the next pseudocode!
+
+// const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+// // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
+// const colors2 = [
+//   "chartreuse",
+//   "indigo",
+//   "periwinkle",
+//   "ochre",
+//   "aquamarine",
+//   "saffron"
+// ]
+// const shuffle = (array) => {
+//   array.shift()
+//     return array
+  
+  // }
 
+
+
+// describe("shuffle", () => {
+//   it("removes the first item from the array and shuffles the remaining content", () => {
+//     expect(shuffle(colors1)).toEqual(expect.arrayContaining(["blue", "green", "yellow", "pink"]))
+//     expect(shuffle(colors1)).not.toEqual(expect.arrayContaining(["purple"]))
+//     expect(shuffle(colors2)).toEqual(expect.arrayContaining([
+//       "indigo",
+//       "periwinkle",
+//       "ochre",
+//       "aquamarine",
+//       "saffron"
+//     ]))
+//     expect(shuffle(colors2)).not.toEqual(expect.arrayContaining([
+//       "chartreuse"
+//     ]))
+//   })
+// })
+// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 // b) Create the function that makes the test pass.
 
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
+//PSEUDOCODE: 
+//Welcome back to some more Pseudocode I hope you enjoyed the first one! In this major event we will be creating a function called voteTracker. voteTracker will take in sets of numbers in the form of upVotes and downVotes parameters where the function will poerform upVotes-downVotes=tallied votes.
+
+
+//**PSEUDOCODE UPDATE: Well i wish i could say i did this myself but i didnt :( What was missing for me is that i CANNOT access the object contents like upvotes and downvotes WITHOUT going through the object its self. */
 // a) Create a test with expect statements for each of the variables provided.
```

Comment:
Not a problem if you need to go back and refactor. I would like to see your initial attempt based on the first pseudocode you had written.